inc/saiswitch.h

```diff
@@ -683,6 +683,18 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID,
 
+    /**
+     * @brief Default SAI Override Virtual Router ID
+     *
+     * Must return #SAI_STATUS_OBJECT_IN_USE when try to delete this VR ID.
+     *
+     * @type sai_object_id_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @default internal
+     */
+    SAI_SWITCH_ATTR_DEFAULT_OVERRIDE_VIRTUAL_ROUTER_ID,
```

Comment:
does this mean that this is equal to default virtual router ? or this is new virtual router and by default there will be 2 of them ?

Reply:
By default there will be two VR

Reply:
what is the motivation to introduce this new override virtual router id?

Reply:
Is it possible to add a capability check to see if the underlying ASIC/SDK supports the feature - using the sai_acl_stage_t SAI_ACL_STAGE_PRE_INGRESS during staging

Reply:
> what is the motivation to introduce this new override virtual router id?

I have added the use case document explaining
https://github.com/opencomputeproject/SAI/pull/1186/commits/bb633a2421a63cb95c7e3d34296c3f422bc642c9

Reply:
> Is it possible to add a capability check to see if the underlying ASIC/SDK supports the feature - using the sai_acl_stage_t SAI_ACL_STAGE_PRE_INGRESS during staging

There are no ACL stages for this change. Override VRF is not supported by SAI adapter will return "ERROR" on the read of this attribute by application.

Reply:
/**
--
  | * @brief List of   actions in sai_acl_action_type_t
  | *
  | * Based on the   ACL capability per stage obtained from the switch
  | * attributes   #SAI_SWITCH_ATTR_ACL_STAGE_INGRESS and #SAI_SWITCH_ATTR_ACL_STAGE_EGRESS
  | * application   should pass the action list if its mandatory per stage.
  | * Pass the   action list if its mandatory per stage.
  | * If its not   mandatory application can either pass the action list
  | * or ignore it.
  | *
  | * @type   sai_s32_list_t sai_acl_action_type_t
  | * @flags   CREATE_ONLY
  | * @default empty
  | */
  | SAI_ACL_TABLE_ATTR_ACL_ACTION_TYPE_LIST,
  |  
This one is for list of actions supported in each stage. You could make a query here for STAGE_PRE_INGRESS and get SET_VRF as action or a NULL list in which case it cannot support this feature. This is for CREATE only, should be extended to get the value to check capability exists.